pkg/controller/user-cluster-controller-manager/flatcar/flatcar_controller.go

```diff
@@ -81,13 +75,30 @@ func (r *Reconciler) Reconcile(ctx context.Context, _ reconcile.Request) (reconc
 		return reconcile.Result{}, nil
 	}
 
-	if err := r.reconcileUpdateOperatorResources(ctx); err != nil {
-		return reconcile.Result{}, fmt.Errorf("failed to reconcile the UpdateOperator resources: %v", err)
+	var nodes corev1.NodeList
+	if err := r.List(ctx, &nodes,
+		ctrlruntimeclient.MatchingLabels{nodelabelerapi.DistributionLabelKey: nodelabelerapi.FlatcarLabelValue},
+	); err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to list nodes: %v", err)
+	}
+
+	if len(nodes.Items) == 0 {
+		if err := r.cleanupUpdateOperatorResources(ctx); err != nil {
+			return reconcile.Result{}, fmt.Errorf("failed to clean up UpdateOperator resources: %v", err)
+		}
+	} else {
+		if err := r.reconcileUpdateOperatorResources(ctx); err != nil {
```

Comment:
Re: schedulable, from the PR notes:

> The check if a node is unschedulable has been removed as both DaemonSet and Deployment ignore ANY taints. Flatcar nodes that are marked as unschedulable can therefore still run the update-agent.

On the deletion timestamp, I agree. Let me add that.

Reply:
> The check if a node is unschedulable has been removed as both DaemonSet and Deployment ignore ANY taints

But the pod will never be created in this case right? So IMO it doesn't make sense to create deployment/daemonset knowing that the pod is not schedulable.

Reply:
We discussed this, it's important to not ignore `unschedulable` Nodes because with a single Flatcar Node, it could be drained and rebooted by the update-operator/agent. That would make the node temporarily unschedulable and trigger the `user-cluster-controller-manager` to remove the operator and agent, something we don't want in that scenario.